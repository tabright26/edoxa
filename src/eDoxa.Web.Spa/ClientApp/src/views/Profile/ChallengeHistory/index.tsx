import React, { FunctionComponent, useEffect } from "react";
import { compose } from "recompose";
import { connect, MapDispatchToProps } from "react-redux";
import { loadChallengeHistory } from "store/actions/challenge";
import ChallengeList from "components/Challenge/List";
import {
  withUserProfileUserId,
  HocUserProfileUserIdStateProps
} from "utils/oidc/containers";

type OwnProps = HocUserProfileUserIdStateProps;

type DispatchProps = { loadChallengeHistory: () => void };

type InnerProps = DispatchProps & OwnProps;

type OutterProps = {};

type Props = InnerProps & OutterProps;

const ProfileChallengeHistory: FunctionComponent<Props> = ({
  loadChallengeHistory,
  userId
}) => {
  useEffect((): void => {
    loadChallengeHistory();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <>
      <h5 className="text-uppercase my-4">CHALLENGE HISTORY</h5>
      <ChallengeList userId={userId} xs="12" sm="12" md="12" lg="12" />
    </>
  );
};

const mapDispatchToProps: MapDispatchToProps<DispatchProps, OwnProps> = (
  dispatch: any
) => {
  return {
    loadChallengeHistory: () => dispatch(loadChallengeHistory())
  };
};

const enhance = compose<InnerProps, OutterProps>(
  withUserProfileUserId,
  connect(null, mapDispatchToProps)
);

export default enhance(ProfileChallengeHistory);
