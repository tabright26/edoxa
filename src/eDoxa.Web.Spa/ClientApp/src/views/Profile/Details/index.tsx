import React, { FunctionComponent } from "react";
import ProfilePanel from "components/Service/Identity/Profile/Panel";
import EmailPanel from "components/Service/Identity/Email/Panel";
import AddressPanel from "components/Service/Identity/Address/Panel";

const ProfileDetails: FunctionComponent = () => (
  <>
    <h5 className="text-uppercase my-4">PROFILE DETAILS</h5>
    <ProfilePanel className="my-4" />
    <EmailPanel className="my-4" />
    <AddressPanel className="my-4" />
  </>
);

export default ProfileDetails;
