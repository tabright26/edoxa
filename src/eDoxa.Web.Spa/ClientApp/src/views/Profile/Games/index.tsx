import React, { FunctionComponent } from "react";

import GameList from "components/Game/List";

const ProfileGames: FunctionComponent = () => {
  return (
    <>
      <h5 className="text-uppercase my-4">GAME CONNECTIONS</h5>
      <GameList />
    </>
  );
};

export default ProfileGames;
