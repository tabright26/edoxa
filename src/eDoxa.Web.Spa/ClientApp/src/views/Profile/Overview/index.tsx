import React, { FunctionComponent } from "react";
import DoxatagPanel from "components/Service/Identity/Doxatag/Panel";

const ProfileOverview: FunctionComponent = () => (
  <>
    <h5 className="text-uppercase my-4">PROFILE OVERVIEW</h5>
    <DoxatagPanel className="my-4" />
  </>
);

export default ProfileOverview;
