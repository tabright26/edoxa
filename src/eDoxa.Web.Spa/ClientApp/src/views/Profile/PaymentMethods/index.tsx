import React, { FunctionComponent } from "react";
import PaymentMethodCardPanel from "components/Service/Payment/Stripe/PaymentMethod/Card/Panel";

const PaymentMethods: FunctionComponent = () => (
  <>
    <h5 className="text-uppercase my-4">PAYMENT METHODS</h5>
    <PaymentMethodCardPanel className="my-4" />
  </>
);

export default PaymentMethods;
