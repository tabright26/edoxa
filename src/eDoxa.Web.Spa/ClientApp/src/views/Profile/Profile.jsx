import React, { Component } from "react";
import { Col, Row, ListGroup, ListGroupItem, TabContent, TabPane } from "reactstrap";

import ProfileDetails from "./Details/Details";

class Profile extends Component {
  state = {
    activeTab: 0
  };

  toggle = activeTab => {
    this.setState({ activeTab });
  };

  render() {
    return (
      <Row>
        <Col xs="4" sm="3" md="2">
          <ListGroup id="list-tab-2" role="tablist" className="my-4">
            <ListGroupItem onClick={() => this.toggle(0)} action active={this.state.activeTab === 0}>
              Profile Overview
            </ListGroupItem>
            <ListGroupItem onClick={() => this.toggle(1)} action active={this.state.activeTab === 1}>
              Profile Details
            </ListGroupItem>
            <ListGroupItem onClick={() => this.toggle(2)} action active={this.state.activeTab === 2}>
              Security
            </ListGroupItem>
            <ListGroupItem onClick={() => this.toggle(3)} action active={this.state.activeTab === 3}>
              Confidentiality
            </ListGroupItem>
            <ListGroupItem onClick={() => this.toggle(4)} action active={this.state.activeTab === 4}>
              Connections
            </ListGroupItem>
            <ListGroupItem onClick={() => this.toggle(5)} action active={this.state.activeTab === 5}>
              Payment Methods
            </ListGroupItem>
            <ListGroupItem onClick={() => this.toggle(6)} action active={this.state.activeTab === 6}>
              Transaction History
            </ListGroupItem>
          </ListGroup>
        </Col>
        <Col xs="12" sm="12" md="7" lg="6">
          <TabContent activeTab={this.state.activeTab} className="border-0 my-4" style={{ background: "none" }}>
            <TabPane tabId={0} className="p-0">
              <h5 className="mb-4">PROFILE OVERVIEW</h5>
            </TabPane>
            <ProfileDetails tabId={1} />
            <TabPane tabId={2} className="p-0">
              <h5 className="mb-4">SECURITY</h5>
            </TabPane>
            <TabPane tabId={3} className="p-0">
              <h5 className="mb-4">CONFIDENTIALITY</h5>
            </TabPane>
            <TabPane tabId={4} className="p-0">
              <h5 className="mb-4">CONNECTIONS</h5>
            </TabPane>
            <TabPane tabId={5} className="p-0">
              <h5 className="mb-4">PAYMENT METHODS</h5>
            </TabPane>
            <TabPane tabId={6} className="p-0">
              <h5 className="mb-4">TRANSACTION HISTORY</h5>
            </TabPane>
          </TabContent>
        </Col>
      </Row>
    );
  }
}

export default Profile;
