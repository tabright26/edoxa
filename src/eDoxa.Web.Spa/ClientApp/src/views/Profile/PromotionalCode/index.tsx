import React, { FunctionComponent } from "react";
import PromotionPanel from "components/Cashier/Promotion/Panel";

const PromotionalCode: FunctionComponent = () => (
  <>
    <h5 className="text-uppercase my-4">PROMOTIONAL CODE</h5>
    <PromotionPanel className="my-4" />
  </>
);

export default PromotionalCode;
