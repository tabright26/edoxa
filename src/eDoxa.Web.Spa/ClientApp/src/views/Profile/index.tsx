import React, { Suspense, FunctionComponent } from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import {
  Card,
  CardHeader,
  Col,
  Row,
  ListGroup,
  ListGroupItem
} from "reactstrap";
import { RouteProps } from "utils/router/types";
import { LinkContainer } from "react-router-bootstrap";
import { Loading } from "components/Shared/Loading";
import {
  getProfileOverviewPath,
  getProfileTransactionHistoryPath,
  getProfilePaymentMethodsPath,
  getProfileGamesPath,
  getProfileSecurityPath,
  getProfileDetailsPath,
  getError404Path,
  getProfilePromotionalCodePath,
  getProfileChallengeHistoryPath
} from "utils/coreui/constants";
import EmailAlert from "components/Email/Alert";

const ProfileOverview = React.lazy(() => import("./Overview"));
const ProfileDetails = React.lazy(() => import("./Details"));
const ProfileSecurity = React.lazy(() => import("./Security"));
const ProfileConnections = React.lazy(() => import("./Games"));
const ProfilePromotionalCode = React.lazy(() => import("./PromotionalCode"));
const ProfilePaymentMethods = React.lazy(() => import("./PaymentMethods"));
const ProfileTransactionHistory = React.lazy(() =>
  import("./TransactionHistory")
);
const ProfileChallengeHistory = React.lazy(() => import("./ChallengeHistory"));

const Profile: FunctionComponent = () => (
  <>
    <Row>
      <Col xs="12" sm="12" md="12" lg="11" xl="9">
        <EmailAlert />
      </Col>
    </Row>
    <Row>
      <Col xs="12" sm="12" md="3" lg="3" xl="2">
        <Card className="mt-4">
          <CardHeader>
            <strong className="text-uppercase">Profile</strong>
          </CardHeader>
          <ListGroup flush>
            <LinkContainer to={getProfileOverviewPath()}>
              <ListGroupItem>Overview</ListGroupItem>
            </LinkContainer>
            <LinkContainer to={getProfileDetailsPath()}>
              <ListGroupItem>Details</ListGroupItem>
            </LinkContainer>
            {process.env.NODE_ENV !== "production" && (
              <LinkContainer to={getProfileSecurityPath()}>
                <ListGroupItem>Security</ListGroupItem>
              </LinkContainer>
            )}
          </ListGroup>
        </Card>
        <Card>
          <CardHeader>
            <strong className="text-uppercase">Arena</strong>
          </CardHeader>
          <ListGroup flush>
            <LinkContainer to={getProfileChallengeHistoryPath()}>
              <ListGroupItem>Challenge History</ListGroupItem>
            </LinkContainer>
          </ListGroup>
        </Card>
        <Card>
          <CardHeader>
            <strong className="text-uppercase">Cashier</strong>
          </CardHeader>
          <ListGroup flush>
            <LinkContainer to={getProfilePaymentMethodsPath()}>
              <ListGroupItem>Payment Methods</ListGroupItem>
            </LinkContainer>
            <LinkContainer to={getProfileTransactionHistoryPath()}>
              <ListGroupItem>Transaction History</ListGroupItem>
            </LinkContainer>
            <LinkContainer to={getProfilePromotionalCodePath()}>
              <ListGroupItem>Promotional Code</ListGroupItem>
            </LinkContainer>
          </ListGroup>
        </Card>
        <Card>
          <CardHeader>
            <strong className="text-uppercase">Connections</strong>
          </CardHeader>
          <ListGroup flush>
            <LinkContainer to={getProfileGamesPath()}>
              <ListGroupItem>Games</ListGroupItem>
            </LinkContainer>
          </ListGroup>
        </Card>
      </Col>
      <Col xs="12" sm="12" md="9" lg="8" xl="7">
        <Suspense fallback={<Loading />}>
          <Switch>
            <Route<RouteProps>
              path={getProfileOverviewPath()}
              exact
              name="Profile Overview"
              component={ProfileOverview}
            />
            <Route<RouteProps>
              path={getProfileDetailsPath()}
              exact
              name="Profile Details"
              component={ProfileDetails}
            />
            {process.env.NODE_ENV !== "production" && (
              <Route<RouteProps>
                path={getProfileSecurityPath()}
                exact
                name="Security"
                component={ProfileSecurity}
              />
            )}
            <Route<RouteProps>
              path={getProfileGamesPath()}
              exact
              name="Connections"
              component={ProfileConnections}
            />
            <Route<RouteProps>
              path={getProfilePaymentMethodsPath()}
              exact
              name="Payment Methods"
              component={ProfilePaymentMethods}
            />
            <Route<RouteProps>
              path={getProfileTransactionHistoryPath()}
              exact
              name="Transaction History"
              component={ProfileTransactionHistory}
            />
            <Route<RouteProps>
              path={getProfilePromotionalCodePath()}
              exact
              name="Promotional Code"
              component={ProfilePromotionalCode}
            />
            <Route<RouteProps>
              path={getProfileChallengeHistoryPath()}
              exact
              name="Challenge History"
              component={ProfileChallengeHistory}
            />
            <Redirect to={getError404Path()} />
          </Switch>
        </Suspense>
      </Col>
    </Row>
  </>
);

export default Profile;
