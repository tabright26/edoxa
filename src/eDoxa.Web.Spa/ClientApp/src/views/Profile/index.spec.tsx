import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import Profile from ".";
import { StripeBankAccountState } from "store/root/payment/stripe/bankAccount/types";
import { StripePaymentMethodsState } from "store/root/payment/stripe/paymentMethod/types";
import { UserTransactionState } from "store/root/user/transactionHistory/types";
import { GamesState } from "store/root/game/types";
import { UserAddressBookState } from "store/root/user/addressBook/types";
import { UserEmailState } from "store/root/user/email/types";
import { UserProfileState } from "store/root/user/profile/types";
import { UserPhoneState } from "store/root/user/phone/types";
import {
  GAME_LEAGUE_OF_LEGENDS,
  Game,
  GameOption,
  GameServiceName,
  TRANSACTION_STATUS_SUCCEEDED,
  TRANSACTION_TYPE_WITHDRAWAL,
  TRANSACTION_TYPE_DEPOSIT,
  TRANSACTION_TYPE_CHARGE,
  CURRENCY_MONEY
} from "types";

it("renders without crashing", () => {
  //Arrange
  const paymentMethods: StripePaymentMethodsState = {
    data: [
      {
        id: "0",
        type: "card",
        card: {
          brand: "visa",
          country: "CA",
          expMonth: 6,
          expYear: 2015,
          last4: "4242"
        }
      }
    ],
    loading: false,
    error: null
  };

  const bankAccount: StripeBankAccountState = {
    data: {
      bankName: "RBC",
      country: "Canada",
      currency: "CAD",
      last4: "4242",
      status: "verified",
      defaultForCurrency: true
    },
    loading: false,
    error: null
  };

  const transactions: UserTransactionState = {
    data: [
      {
        timestamp: 111111,
        currency: CURRENCY_MONEY,
        amount: 100,
        description: "test",
        type: TRANSACTION_TYPE_CHARGE,
        status: TRANSACTION_STATUS_SUCCEEDED,
        id: "1"
      },
      {
        timestamp: 222222,
        currency: CURRENCY_MONEY,
        amount: 100,
        description: "test",
        type: TRANSACTION_TYPE_DEPOSIT,
        status: TRANSACTION_STATUS_SUCCEEDED,
        id: "2"
      },
      {
        timestamp: 333333,
        currency: CURRENCY_MONEY,
        amount: 100,
        description: "test",
        type: TRANSACTION_TYPE_WITHDRAWAL,
        status: TRANSACTION_STATUS_SUCCEEDED,
        id: "3"
      }
    ],
    loading: false,
    error: null
  };

  const games: GamesState = {
    data: new Map<Game, GameOption>(),
    loading: false,
    error: null
  };

  games.data.set(GAME_LEAGUE_OF_LEGENDS, {
    name: GAME_LEAGUE_OF_LEGENDS,
    displayName: "League of Legends",
    displayed: true,
    verified: true,
    instructions: null,
    services: new Map<GameServiceName, boolean>()
  });

  const addressBook: UserAddressBookState = {
    data: [
      {
        country: "Canada",
        line1: "test address",
        city: "Montreal",
        id: "1"
      },
      {
        country: "Canada",
        line1: "test address",
        city: "Quebec",
        id: "2"
      }
    ],
    loading: false,
    error: null
  };

  const email: UserEmailState = {
    data: { address: "gabriel@edoxa.gg", verified: true },
    loading: false,
    error: null
  };

  const informations: UserProfileState = {
    data: {
      firstName: "Gabriel",
      lastName: "Roy",
      gender: "Male",
      dob: { year: 1995, month: 8, day: 4 }
    },
    loading: false,
    error: null
  };

  const phone: UserPhoneState = {
    data: { number: "123456789", verified: true },
    loading: false,
    error: null
  };

  const store: any = {
    getState: () => {
      return {
        root: {
          user: {
            transactions,
            games,
            addressBook,
            email,
            informations,
            phone
          },
          payment: {
            stripe: {
              bankAccount,
              paymentMethods
            }
          }
        }
      };
    },
    dispatch: action => {},
    subscribe: () => {}
  };

  //Act
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Profile
            match={{ params: "userId", isExact: false, path: "", url: "" }}
            history={null}
            location={null}
          />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();

  //Assert
  expect(tree).toMatchSnapshot();
});
