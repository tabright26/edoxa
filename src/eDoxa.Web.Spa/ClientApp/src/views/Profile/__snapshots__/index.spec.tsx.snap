// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="row"
>
  <div
    className="col-12 col-sm-12 col-md-4 col-lg-3 col-xl-2"
  >
    <div
      className="mt-4 card"
    >
      <div
        className="bg-gray-900 card-header"
      >
        <strong
          className="text-uppercase"
        >
          Profile
        </strong>
      </div>
      <ul
        className="list-group list-group-flush"
      >
        <li
          className="list-group-item"
          href="/profile/overview"
          onClick={[Function]}
        >
          Overview
        </li>
        <li
          className="list-group-item"
          href="/profile/details"
          onClick={[Function]}
        >
          Details
        </li>
        <li
          className="list-group-item"
          href="/profile/security"
          onClick={[Function]}
        >
          Security
        </li>
      </ul>
    </div>
    <div
      className="card"
    >
      <div
        className="bg-gray-900 card-header"
      >
        <strong
          className="text-uppercase"
        >
          Arena
        </strong>
      </div>
      <ul
        className="list-group list-group-flush"
      >
        <li
          className="list-group-item"
          href="/profile/challenge-history"
          onClick={[Function]}
        >
          Challenge History
        </li>
      </ul>
    </div>
    <div
      className="card"
    >
      <div
        className="bg-gray-900 card-header"
      >
        <strong
          className="text-uppercase"
        >
          Cashier
        </strong>
      </div>
      <ul
        className="list-group list-group-flush"
      >
        <li
          className="list-group-item"
          href="/profile/payment-methods"
          onClick={[Function]}
        >
          Payment Methods
        </li>
        <li
          className="list-group-item"
          href="/profile/transaction-history"
          onClick={[Function]}
        >
          Transaction History
        </li>
        <li
          className="list-group-item"
          href="/profile/promotional-code"
          onClick={[Function]}
        >
          Promotional Code
        </li>
      </ul>
    </div>
    <div
      className="card"
    >
      <div
        className="bg-gray-900 card-header"
      >
        <strong
          className="text-uppercase"
        >
          Connections
        </strong>
      </div>
      <ul
        className="list-group list-group-flush"
      >
        <li
          className="list-group-item"
          href="/profile/games"
          onClick={[Function]}
        >
          Games
        </li>
      </ul>
    </div>
  </div>
  <div
    className="col-12 col-sm-12 col-md-8 col-lg-7 col-xl-6"
  />
</div>
`;
